### PR TITLE
Command.Deploy: add checkSystemKnownHosts option to ob deploy init ref: #569

### PR DESCRIPTION
**Pull request by romefeller**
This PR is the new version of PR 569.

https://github.com/obsidiansystems/obelisk/pull/569/

### CODE REVIEW

File: lib/command/src/Obelisk/Command.hs [hunk 1]
```diff
@@ -185,6 +186,14 @@ data DeployCommand
   | DeployCommand_Update
   deriving Show
 
+--TODO: Result should provide normalised path and also original user input for error reporting.
+thunkDirectoryParser :: Parser FilePath
```
Comment (madeline-os) [inline lib/command/src/Obelisk/Command.hs:190]: Please add a haddock and remove this TODO. make an issue for it after we merge this PR

File: lib/command/src/Obelisk/Command/Deploy.hs [hunk 2]
```diff
@@ -313,6 +320,24 @@ readDeployConfig :: MonadObelisk m => FilePath -> FilePath -> m String
 readDeployConfig deployDir fname = liftIO $ do
   fmap (T.unpack . T.strip) $ T.readFile $ deployDir </> fname
 
+lookupKnownHosts :: MonadObelisk m  => String -> m [BS.ByteString]
```
Comment (madeline-os) [inline lib/command/src/Obelisk/Command/Deploy.hs:323]: Haddock

File: lib/command/src/Obelisk/Command/Deploy.hs [hunk 3]
```diff
@@ -313,6 +320,24 @@ readDeployConfig :: MonadObelisk m => FilePath -> FilePath -> m String
...
+  fmap filterComments $ readCreateProcessWithExitCode $ proc "ssh-keygen" ["-F", hostName]
+   where
+     filterComments (exitCode, out, _) =
+       if exitCode /= ExitSuccess || null out
+         then []
+         else
+           -- ssh-keygen prints the following above each result it finds: "# Host <hostname> found: line <lineno>"
+           filter (not . C.isPrefixOf "# Host") $ C.lines $ C.pack out
+
+addKnownHostFromEnv :: MonadObelisk m => String -> FilePath -> m ()
```
Comment (madeline-os) [inline lib/command/src/Obelisk/Command/Deploy.hs:334]: Haddock

**Review by madeline-os — CHANGES_REQUESTED** [review]
Comment fixes requested

File: lib/command/src/Obelisk/Command/Deploy.hs (hunk 3 above)
Comment (romefeller) [reply lib/command/src/Obelisk/Command/Deploy.hs]: Done

**Review by madeline-os — APPROVED** [review]

File: lib/command/src/Obelisk/Command/Deploy.hs [hunk 4]
```diff
@@ -313,6 +321,35 @@ readDeployConfig :: MonadObelisk m => FilePath -> FilePath -> m String
...
   fmap (T.unpack . T.strip) $ T.readFile $ deployDir </> fname
 
+-- | Lookup known hosts using ssh-keygen command
+lookupKnownHosts :: MonadObelisk m  
+                 => String 
+                 -- ^ the host name
+                 -> m [BS.ByteString]
+                 -- ^ obtained hosts
+lookupKnownHosts hostName =
+  fmap filterComments $ readCreateProcessWithExitCode $ proc "ssh-keygen" ["-F", hostName]
```
Comment (alexfmpe) [inline lib/command/src/Obelisk/Command/Deploy.hs:331]: https://github.com/obsidiansystems/obelisk/pull/569#discussion_r357864129

Comment (madeline-os) [reply lib/command/src/Obelisk/Command/Deploy.hs:331]: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

Comment (madeline-os) [reply lib/command/src/Obelisk/Command/Deploy.hs:331]: thank yo ualex

**Pull request merged**